pkg/eventprocessor/impl/eventprocessor.go

```diff
@@ -139,6 +142,7 @@ func (ep *EventProcessor) startDaemon() error {
 		return fmt.Errorf("get last executed block number: %s", err)
 	}
 	ep.mLastProcessedHeight.Store(fromHeight)
+	ep.nextHashCalcBlockNumber = nextMultipleOf(fromHeight, ep.config.HashCalcStep)
```

Comment:
When we start `EventProcessor` we have to figure out the next block which it will do the hash calculation. It is the next block greater than `fromHeight` that is multiple of  `ep.config.HashCalcStep`.